code/ai/ailua.cpp

```diff
@@ -48,6 +48,149 @@ const player_order_lua* ai_lua_find_player_order(int sexp_op) {
 		return &aiLuaOrder->second;
 }
 
+int ai_lua_get_num_general_orders() {
+	int count = 0;
+	
+	for (auto order : Lua_player_orders) {
+		if (order.second.generalOrder) {
+			count++;
+		}
+	}
+
+	return count;
+}
+
+SCP_vector<SCP_string> ai_lua_get_general_order_categories(bool enabled_only)
+{
+	SCP_vector<SCP_string> list;
+
+	for (auto order : Lua_player_orders) {
+		SCP_string cat = order.second.category;
+
+		// If it's not a general order then move on
+		if (!order.second.generalOrder) {
+			continue;
+		}
+
+		// If the order is not enabled then move on
+		if (enabled_only && !order.second.cur_enabled) {
+			continue;
+		}
+
+		Assertion((cat.length() > 0), "Lua general orders category name must be longer than 0 characters!");
+
+		for (size_t i = 0; i <= list.size(); i++) {
+
+			// reached the end of the list, so add it
+			if (i == list.size()) {
+				list.push_back(cat);
+				break;
+			}
+
+			// if it's already in the list then don't add it
+			if (cat == list[i]) {
+				break;
+			}
+		}
+	}
+
+	return list;
+}
+
+SCP_vector<SCP_string> ai_lua_get_enabled_general_orders_by_category(SCP_string cat) {
```

Comment:
As this is also is just a read-only thing, it should be a const SCP_string& as well